### PR TITLE
feat: cySync log improvements

### PR DESCRIPTION
**Pull request by jhasuraj01**
- dialogs navigation
- pages navigation
- button clicks
- dropdown changes

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6931 CySync logs improvement for better diagnosis of user issues](https://app.clickup.com/t/9002019994/PRF-6931)

File: packages/cysync-core/src/actions/wallet/syncWallets.ts [hunk 1]
```diff
@@ -23,18 +23,31 @@ const syncWalletsWithConnectedDevice = async (
...
-      connection.status !== DeviceConnectionStatus.CONNECTED ||
-      !connection.isAuthenticated
-    ) {
+    if (!connection?.connection) {
+      logger.warn('Wallet Sync Failed: Connection is not set up');
+      return undefined;
+    }
+
+    if (!connection.isMain) {
+      logger.warn('Wallet Sync Failed: Main App is not open');
```
Comment (irshadCypherock) [inline packages/cysync-core/src/actions/wallet/syncWallets.ts]:
```suggestion
      logger.warn('Wallet Sync Abort: Device is not on main');
```

File: packages/cysync-core/src/actions/wallet/syncWallets.ts [hunk 2]
```diff
@@ -23,18 +23,31 @@ const syncWalletsWithConnectedDevice = async (
...
+
   try {
-    if (
-      !connection?.connection ||
-      !connection.isMain ||
-      connection.status !== DeviceConnectionStatus.CONNECTED ||
-      !connection.isAuthenticated
-    ) {
+    if (!connection?.connection) {
+      logger.warn('Wallet Sync Failed: Connection is not set up');
```
Comment (irshadCypherock) [inline packages/cysync-core/src/actions/wallet/syncWallets.ts]:
```suggestion
      logger.warn('Wallet Sync Abort: Connection is not set up');
```

File: packages/cysync-core/src/actions/wallet/syncWallets.ts [hunk 3]
```diff
@@ -23,18 +23,31 @@ const syncWalletsWithConnectedDevice = async (
...
+      return undefined;
+    }
+
+    if (!connection.isMain) {
+      logger.warn('Wallet Sync Failed: Main App is not open');
+      return undefined;
+    }
+
+    if (connection.status !== DeviceConnectionStatus.CONNECTED) {
+      logger.warn('Wallet Sync Failed: Device is not connected');
```
Comment (irshadCypherock) [inline packages/cysync-core/src/actions/wallet/syncWallets.ts]:
```suggestion
      logger.warn('Wallet Sync Abort: Device is not connected');
```

File: packages/cysync-core/src/actions/wallet/syncWallets.ts [hunk 4]
```diff
@@ -23,18 +23,31 @@ const syncWalletsWithConnectedDevice = async (
...
+      return undefined;
+    }
+
+    if (!connection.isAuthenticated) {
+      logger.warn('Wallet Sync Failed: Device is not authenticated');
```
Comment (irshadCypherock) [inline packages/cysync-core/src/actions/wallet/syncWallets.ts]:
```suggestion
      logger.warn('Wallet Sync Abort: Device is not authenticated');
```

File: packages/cysync-core/src/actions/wallet/syncWallets.ts [hunk 5]
```diff
@@ -23,18 +23,31 @@ const syncWalletsWithConnectedDevice = async (
...
       return undefined;
     }
 
     if (!connection.serial) {
-      logger.warn('No serial found for connected device');
+      logger.warn('Wallet Sync Failed: No serial found for connected device');
```
Comment (irshadCypherock) [inline packages/cysync-core/src/actions/wallet/syncWallets.ts]:
```suggestion
      logger.warn('Wallet Sync Abort: No serial found for connected device');
```

File: packages/cysync-core/src/components/Graph.tsx [hunk 6]
```diff
@@ -35,6 +36,17 @@ export const Graph: React.FC<GraphProps> = ({
     formatGraphAmountDisplay,
   } = useGraph({ selectedWallet, accountId, assetId, parentAssetId });
 
+  const handleWalletChangeProxy = useMemo((): typeof handleWalletChange => {
```
Comment (irshadCypherock) [inline packages/cysync-core/src/components/Graph.tsx:39]:
```suggestion
  const handleWalletChangeProxy = useCallback((): typeof handleWalletChange => {
```

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/cysync-core/src/components/Graph.tsx (hunk 6 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/components/Graph.tsx:39]: Resolved as per discussion on slack.

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**